### PR TITLE
Hash & compress test

### PR DESCRIPTION
**Pull request by iwanbk**

### CODE REVIEW

File: compress/compress_test.go [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+	"crypto/rand"
+	"io/ioutil"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+// TestRoundTrip tests that gzipping and then gunzipping is the identity
+// function.
+func TestRoundTrip(t *testing.T) {
```
Comment (zaibon) [inline compress/compress_test.go]: Can you use the sub test feature here, so we can have a nicer output with names when running the tests

File: compress/compress_test.go [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+	assert.Nil(t, err)
+
+	err = r.Close()
+	assert.Nil(t, err)
+
+	// compare decompressed with original payload
+	assert.Equal(t, payload, b)
+}
+
+func TestWriterReset(t *testing.T) {
```
Comment (zaibon) [inline compress/compress_test.go]: same comment, sub test would be nice

File: hash/hash_test.go [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+	"crypto/md5"
+	"crypto/rand"
+	"crypto/sha256"
+	"testing"
+
+	"github.com/minio/blake2b-simd"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestHash(t *testing.T) {
```
Comment (zaibon) [inline hash/hash_test.go]: same here.

Comment (iwanbk) [inline hash/hash_test.go]: Sure. Done

**Pull request merged**